src/components/projects/project-edit-form/project-edit-container/project-edit-container.tsx

```diff
@@ -41,10 +53,28 @@ const ProjectEditContainer: React.FC<EditFormContainerProps> = ({ wbsNum, proj,
   const updateBom = (url: string | undefined) => setBom(url);
   const updateGDrive = (url: string | undefined) => setGDrive(url);
 
-  const [goals, setGoals] = useState(proj!.goals.map((goal) => goal.detail));
-  const [features, setFeatures] = useState(proj!.features.map((feature) => feature.detail));
-  const [otherConstraints, setOther] = useState(
-    proj!.otherConstraints.map((constraint) => constraint.detail)
+  const { mutateAsync } = useEditSingleProject();
+
+  const [goals, setGoals] = useState<{ id?: number; detail: string }[]>(
+    proj!.goals
+      .filter((goal) => !goal.dateDeleted)
+      .map((goal) => {
+        return { id: goal.id, detail: goal.detail };
+      })
+  );
+  const [features, setFeatures] = useState<{ id?: number; detail: string }[]>(
+    proj!.features
+      .filter((feature) => !feature.dateDeleted)
+      .map((feature) => {
+        return { id: feature.id, detail: feature.detail };
+      })
+  );
+  const [otherConstraints, setOther] = useState<{ id?: number; detail: string }[]>(
+    proj!.otherConstraints
+      .filter((constraint) => !constraint.dateDeleted)
+      .map((constraint) => {
+        return { id: constraint.id, detail: constraint.detail };
```

Comment:
Abstraction? We do this 3 times? Just thinking out loud

Reply:
true!